{"pageProps":{"questions":[{"id":"KPQ7kTIRSDOrzG4h3Cbo","isMC":true,"topic":"1","answer":"C","unix_timestamp":1584046500,"answer_images":[],"question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0004100002.png"],"answer_description":"","choices":{"A":"sequential","B":"named","D":"keyword","C":"positional"},"question_id":101,"answer_ET":"C","discussion":[{"content":"Thee answer is C, its positional.","upvote_count":"25","timestamp":"1584046500.0","poster":"Avidulam","comment_id":"63241"},{"upvote_count":"6","comment_id":"66720","poster":"Rajdeep","timestamp":"1584842640.0","content":"Its C- Positional"},{"poster":"seaverick","comment_id":"1140042","content":"Selected Answer: C\nhttps://builtin.com/software-engineering-perspectives/arguments-in-python\n\nKeyword arguments should follow positional arguments only.\n\ndef add(a,b,c):\n return (a+b+c)\nThe above function can be called in two ways:\n\nFirst, during the function call, all arguments are given as positional arguments. Values passed through arguments are passed to parameters by their position. 10 is assigned to a, 20 is assigned to b and 30 is assigned to c.\n\nprint (add(10,20,30))\n#Output:60\nThe second way is by giving a mix of positional and keyword arguments. Keyword arguments should always follow positional arguments.\n\nprint (add(10,c=30,b=20))\n#Output:60","upvote_count":"1","timestamp":"1707046200.0"},{"upvote_count":"1","poster":"SheilaM","timestamp":"1678178880.0","content":"C , itâ€™s positional","comment_id":"831676"},{"timestamp":"1669944240.0","upvote_count":"1","poster":"ekossov","content":"The answer is correct. it is positional","comment_id":"733304"},{"poster":"macxsz","upvote_count":"1","timestamp":"1651583460.0","content":"Selected Answer: C\nC. positional","comment_id":"596379"},{"comment_id":"172854","content":"Its positional","poster":"Spectra","upvote_count":"2","timestamp":"1599156840.0"}],"answers_community":["C (100%)"],"url":"https://www.examtopics.com/discussions/python-institute/view/16436-exam-pcap-topic-1-question-62-discussion/","question_text":"A method for passing the arguments used by the following snippet is called:\n//IMG//","exam_id":192,"timestamp":"2020-03-12 21:55:00"},{"id":"GkeXyldCBeKxKIc8Kiza","url":"https://www.examtopics.com/discussions/python-institute/view/67128-exam-pcap-topic-1-question-63-discussion/","answers_community":["B (100%)"],"answer_images":["https://www.examtopics.com/assets/media/exam-media/04235/0004300001.png"],"question_id":102,"discussion":[{"content":"Technically all choices are wrong since the code will cause an IndentationError just like some other question here but B if lined up correctly.","comment_id":"950125","poster":"CC_DC","upvote_count":"2","timestamp":"1689189300.0"},{"upvote_count":"2","content":"Selected Answer: B\nif indentation was correct:\nB. print 1 2","timestamp":"1651583640.0","poster":"macxsz","comment_id":"596382"},{"content":"There is no correct answer because of IndentationError.\nAssume this code is right indent the answer will be B. print 1 2","upvote_count":"2","comment_id":"590849","poster":"Norasit","timestamp":"1650765360.0"},{"comment_id":"568352","comments":[{"comment_id":"568353","timestamp":"1647346380.0","upvote_count":"1","poster":"rocky48","content":"Otherwise it will cause a runtime exception."}],"upvote_count":"1","poster":"rocky48","content":"If the indentation of Line 2 & 3 is correct, then 1, 2 is the correct answer.","timestamp":"1647346380.0"},{"poster":"Backy","timestamp":"1644428460.0","content":"// This is equivalent to \nfor i in range(1,3):\n print(i, end=' ')\n\n// Any function that includes \"yield\" is called a generator and it works like range(), so that you can use them both in a loop like \"for\", but there are differences, e.g. you can use a generator inside next() but you cannot use range() inside next() because range() is stateless","upvote_count":"2","comment_id":"543967"},{"timestamp":"1638491160.0","poster":"ruydrigo","comments":[{"timestamp":"1638646500.0","comment_id":"493919","upvote_count":"3","poster":"techdawgs","content":"With f(2), n = 2, so the range will be range(1, {2+1}) or range(1,3). The iterations of i would then be 1 and then 2 with yield sending each iteration back to \"for i in f(2)\". Hope this explains it."}],"content":"I run it but I don't understand why yield return 1,2","upvote_count":"1","comment_id":"492863"}],"answer":"B","question_text":"What is the expected behavior of the following code?\n//IMG//\n\n\nIt will -","topic":"1","exam_id":192,"isMC":true,"answer_ET":"B","timestamp":"2021-12-03 01:26:00","unix_timestamp":1638491160,"answer_description":"","choices":{"A":"print 2 1","C":"cause a runtime exception","B":"print 1 2","D":"print <generator object f at (some hex digits)>"},"question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0004200001.png"]},{"id":"1wpMXnWRvwqKnqCHQhgV","timestamp":"2020-03-12 21:59:00","answers_community":["D (53%)","B (47%)"],"question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0004300002.png"],"discussion":[{"upvote_count":"16","comments":[{"upvote_count":"10","comment_id":"90722","poster":"prak","content":"1st is invalid parameter, so the code will issue syntax error.","timestamp":"1589734860.0"},{"content":"its a bit tricky this one. lst or 1st?","upvote_count":"2","poster":"jasonvolta","comment_id":"99549","timestamp":"1590952800.0"},{"content":"there is a space between two equal signs, so it is an error","timestamp":"1593300600.0","comments":[{"timestamp":"1630836780.0","content":"There is a space in the range function too so... we'll never know","upvote_count":"1","poster":"catarata","comment_id":"439634"}],"poster":"myname80","comment_id":"121479","upvote_count":"2"}],"poster":"Avidulam","comment_id":"63243","content":"Length will be 3, the answer is D","timestamp":"1584046740.0"},{"comment_id":"284350","poster":"wprogrammer","content":"The answer B is OK because the name of the variable \"1st\" is incorrect. Variables names can contain numbers but not at the begin.","timestamp":"1612552020.0","upvote_count":"13"},{"upvote_count":"1","comment_id":"1176540","timestamp":"1710769380.0","content":"Selected Answer: D\n>>> lst = [x for x in range(5)]\n>>> lst\n[0, 1, 2, 3, 4]\n>>> lst = list(filter(lambda x: x % 2 == 0, lst))\n>>> lst\n[0, 2, 4]\n>>> print(len(lst))\n3","poster":"julmarcas"},{"content":"print(lst)\n[0, 2, 4]\n\nThe answer: D","timestamp":"1710580920.0","upvote_count":"1","poster":"peypa","comment_id":"1174877"},{"comment_id":"1114429","upvote_count":"1","content":"Selected Answer: D\n>>> print(lst)\n[0, 2, 4]\n>>>","timestamp":"1704450180.0","poster":"blaze056"},{"content":"Selected Answer: D\nIf you ignore the fact the 1st is an invalid variable name and the spacing is all wonky, then the answer is D. It might be lst but its hard to know or sure because formating on this site is a nightmare for programming questions. \nAnyway, if we ignore that, then we evaluate each item in the list [0,1,2,3,4] according to the lamba function x % 2 == 0, where x is each subsequent item in that list. If they evaluate to true, then that item is allowed to pass the filter into the new list (that is created by the list() function).\nRemember that modulo % returns the remainder of the division (regardless of which way you go)\n0 % 2 returns 0\n1 % 2 returns 1\n2 % 2 returns 0\n3 % 2 returns 1\n4 % 2 returns 0\nonly the items that evaluated to 0 will pass the filter.\nThats [0,2,4] with a length of 3","comment_id":"1089722","timestamp":"1701892260.0","poster":"TheFivePips","upvote_count":"3"},{"content":"It is a format issue by examtopics, read 1 as l (letter) everywhere in code.. so, D is the correct answer for this!","timestamp":"1692044040.0","comment_id":"981081","poster":"Mikku123","upvote_count":"1"},{"timestamp":"1689077280.0","comment_id":"948938","poster":"Administrator_Of_Silly_Walks","content":"Selected Answer: B\nB is the correct answer because there is no list named \"list\".","upvote_count":"2","comments":[{"content":"Also, you can't start a lists name with a number.","upvote_count":"1","timestamp":"1689077400.0","comment_id":"948940","poster":"Administrator_Of_Silly_Walks"}]},{"comment_id":"925787","upvote_count":"3","comments":[{"poster":"CC_DC","timestamp":"1689189540.0","upvote_count":"1","content":"True. The assumption is that that should be an letter \"l\" given the choices. If the real exam had '1st' then B would be wrong since this code would not even compile.","comment_id":"950131"}],"poster":"Ello2023","content":"Selected Answer: B\nB.\nyou can not have a variable name where the initial character is a number '1st' it could be 'st1'.\nThe equal sign has a gap which is not recognised in python = =, it should have been ==","timestamp":"1686981360.0"},{"content":"Selected Answer: D\nthe answer is D","comment_id":"717204","timestamp":"1668330960.0","poster":"jaimebb","upvote_count":"1"},{"poster":"Jnanada","comment_id":"648591","content":"Correct Answer is D. If there is space, then it wouldn't give runtime error, it will give compile time error.","timestamp":"1660850460.0","upvote_count":"1"},{"poster":"Profstevie","timestamp":"1660822920.0","comment_id":"648416","content":"It is not B because either error is synthax or compiler. It is never runtime error","upvote_count":"1"},{"timestamp":"1658864640.0","upvote_count":"2","content":"The answer is #B. The code will cause a runtime exception.\nThe actual error is:\n#TypeError: object of type 'filter' has no len()\nwe cant get the len of a filter object so we can not \nprint(len(list) \nThe only answer is B and noting else.","poster":"Janpcap123","comments":[{"timestamp":"1659834660.0","content":"the answer is B but due to space between = =, len can be applied to the filter.","upvote_count":"1","poster":"Gaddipati","comment_id":"643553"}],"comment_id":"637607"},{"comment_id":"606953","timestamp":"1653437340.0","poster":"palagus","upvote_count":"1","content":"Selected Answer: D\nIt is D. len(lst) = 3"},{"upvote_count":"1","timestamp":"1651583700.0","comment_id":"596383","content":"Selected Answer: D\nassuming code is correct:\nD. 3","poster":"macxsz"},{"comment_id":"561991","upvote_count":"2","content":"Selected Answer: B\nWhen you try to run this code, it errors, correct answer is B","timestamp":"1646568840.0","poster":"TheNetworkStudent"},{"upvote_count":"2","content":"1st is not valid variable name. It will give runtime error","poster":"AtulVSharma","comment_id":"466208","timestamp":"1634913780.0"},{"comment_id":"457206","timestamp":"1633360140.0","upvote_count":"2","poster":"Nativo","content":"0 Module 2 == 0 \n2 Module 2 == 0\n4 Module 2 == 0\nAnswer is D"},{"poster":"FR99","upvote_count":"4","timestamp":"1604228880.0","comment_id":"210386","content":"For this question, we need to know that the equal operator is correct: == (correct) vs = = (wrong) and the variable name is compliant: lst (correct) vs 1st (wrong)\nIn such case, the answer for this question is: 3"},{"poster":"maritadev","comment_id":"135724","content":"if between the equals signs, isn't space, the ans correct is 3. so, is D.","timestamp":"1594818420.0","upvote_count":"1"},{"upvote_count":"2","timestamp":"1593309900.0","content":"its pretty obvious that is a one and not a lowercase L. which would cause an error, a variable declaration cannot start with a number.","comment_id":"121537","poster":"WillyNilly69"}],"answer_description":"","url":"https://www.examtopics.com/discussions/python-institute/view/16437-exam-pcap-topic-1-question-64-discussion/","exam_id":192,"answer_ET":"D","choices":{"B":"The code will cause a runtime exception","C":"1","D":"3","A":"2"},"question_id":103,"unix_timestamp":1584046740,"question_text":"What is the expected output of the following code?\n//IMG//","topic":"1","answer_images":[],"answer":"D","isMC":true},{"id":"5LMfIOUg2oVhGX9kUXNW","discussion":[{"content":"the answer is B, print has syntax error","poster":"Avidulam","comment_id":"63246","upvote_count":"15","timestamp":"1584046800.0"},{"poster":"nmg","comment_id":"91681","content":"B\n1 returns int\n2 return None\nyou can't add them","timestamp":"1589845320.0","upvote_count":"13"},{"comment_id":"1140123","poster":"Damon54","upvote_count":"1","content":"What is the expected output of the following code ?\n\ndef unclear (x):\n if x % 2 == 1:\n return 0\n\nprint ( unclear (1) + unclear (2))\nSo, the provided code will result in a type error during execution due to the sum of values of different types. If you want to get a valid result, you should modify the unclear function to return a value other than None when x is even, for example, returning 1 in that case.","timestamp":"1707052320.0"},{"poster":"Ello2023","comment_id":"925790","timestamp":"1686981780.0","content":"Selected Answer: B\nB \n'==' has space in between \nprint brackets starts with end bracket ')'","upvote_count":"1"},{"content":"The expected behavior of the given code is as follows:\n\nThe function unclear(x) takes an integer x as input and checks if it is an odd number. If x is odd, the function returns 0, otherwise it does not return anything and continues to execute.\n\nIn the code, the function unclear(1) is called first with the argument 1. Since 1 is an odd number, the function returns 0.\n\nThen, the function unclear(2) is called with the argument 2. Since 2 is an even number, the function does not return anything and continues to execute.\n\nFinally, the result of unclear(1) + unclear(2) is evaluated. Since unclear(1) returns 0 and unclear(2) does not return anything, the result of unclear(1) + unclear(2) is 0 + None, which will raise a TypeError at runtime.\n\nTherefore, the expected behavior of the code is to print an error message, specifically a TypeError.","timestamp":"1679229600.0","upvote_count":"3","comment_id":"843743","comments":[{"content":"Where does it check if x is odd or even number?","upvote_count":"1","comment_id":"936498","timestamp":"1687948860.0","poster":"Ello2023"}],"poster":"andr3"},{"timestamp":"1658865060.0","comment_id":"637609","upvote_count":"3","poster":"Janpcap123","content":"if the ) after print the answer is B cause a runtime error, if the print statement is syntactically correct the answer would be:\nTypeError: unsupported operand type(s) for +: 'int' and 'NoneType' \nTypeError: unsupported operand type(s) for +: 'int' and 'NoneType'"},{"timestamp":"1651583760.0","content":"Selected Answer: B\nB. cause a runtime exception","poster":"macxsz","comment_id":"596384","upvote_count":"1"},{"upvote_count":"1","poster":"rocky48","content":"Selected Answer: B\nSyntax Error bcoz of the print statement.","comment_id":"589751","timestamp":"1650605400.0"},{"upvote_count":"1","poster":"karans","comment_id":"521722","content":"The answer is B because the print statement isn,t in the correct format\nCorrect format-print()\nThe format they have given - print)","timestamp":"1641928560.0","comments":[{"poster":"karans","comment_id":"521723","content":"Isn't*","timestamp":"1641928620.0","upvote_count":"1"}]},{"timestamp":"1590003780.0","content":"Option B.\nExecuting the program gives the following output:\nprint(unclear(1)+unclear(2)) \nTypeError: unsupported operand type(s) for +: 'int' and 'NoneTy\npe'","comment_id":"92952","poster":"Divya1008","upvote_count":"5"}],"topic":"1","exam_id":192,"answer_images":[],"choices":{"C":"prints 3","D":"print an empty line","B":"cause a runtime exception","A":"print 0"},"isMC":true,"question_id":104,"answer_ET":"B","timestamp":"2020-03-12 22:00:00","answer":"B","question_text":"What is the expected behavior of the following code?\n//IMG//\n\nIt will:","url":"https://www.examtopics.com/discussions/python-institute/view/16438-exam-pcap-topic-1-question-65-discussion/","question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0004400001.png"],"answers_community":["B (100%)"],"unix_timestamp":1584046800,"answer_description":""},{"id":"AkfB0GGGUfbP8PsHIPWE","timestamp":"2022-03-28 08:45:00","answer_ET":"A","choices":{"D":"the class component has to be a method","B":"the class component has to be an instance variable","C":"the class component has to be a class variable","A":"the class component's name will be mangled"},"answer_images":[],"answers_community":["A (100%)"],"topic":"1","question_images":[],"isMC":true,"question_id":105,"answer":"A","exam_id":192,"answer_description":"Reference:\nhttps://hackernoon.com/understanding-the-underscore-of-python-309d1a029edc","question_text":"If any of a class's components has a name that starts with two underscores (___), then:","discussion":[{"comment_id":"743155","content":"A\n\nthe double underscore (__) prefix is used to denote so-called \"private\" class members, i.e. class attributes or methods that are intended to be used only within the class or its subclasses, and should not be accessed directly from outside the class. When a class member is given a double underscore prefix, the interpreter automatically mangles the name of the member by adding a prefix and a suffix to it. This is done to prevent name clashes between class members with the same name but defined in different classes.","upvote_count":"4","timestamp":"1670869080.0","poster":"Adeshina"},{"comment_id":"637611","timestamp":"1658865120.0","upvote_count":"1","poster":"Janpcap123","content":"Absolutely A is the answer."},{"timestamp":"1651583760.0","content":"Selected Answer: A\nA. the class component's name will be mangled","comment_id":"596385","upvote_count":"1","poster":"macxsz"},{"timestamp":"1648449900.0","content":"Selected Answer: A\nI think answer is A","upvote_count":"1","comment_id":"576661","poster":"AlMargoi"}],"url":"https://www.examtopics.com/discussions/python-institute/view/73132-exam-pcap-topic-1-question-66-discussion/","unix_timestamp":1648449900}],"exam":{"name":"PCAP","id":192,"provider":"Python-institute","lastUpdated":"12 Apr 2025","isBeta":false,"numberOfQuestions":141,"isImplemented":true,"isMCOnly":false},"currentPage":21},"__N_SSP":true}